src/Cake.Core/IO/ProcessRunner.cs

```diff
@@ -69,18 +69,19 @@ public IProcess Start(FilePath filePath, ProcessSettings settings)
 
             // Get the working directory.
             var workingDirectory = settings.WorkingDirectory ?? _environment.WorkingDirectory;
-            settings.WorkingDirectory = workingDirectory.MakeAbsolute(_environment);
 
             // Create the process start info.
             var info = new ProcessStartInfo(fileName)
             {
                 Arguments = arguments.Render(),
-                WorkingDirectory = workingDirectory.FullPath,
+                WorkingDirectory = settings.WorkingDirectory != null ? workingDirectory.FullPath : null,
```

Comment:
I think we should move the assignment and only assign if `settings.WorkingDirectory` is not null so we don't `touch it` unless we want to set it.